Added calculation of launch angles of the RF trajectory from the source vertex.

FlahertyJustinC opened the pull request:
Implemented calculation and storage of launch angles of the RF signal from the source vertex to work in conjunction with [AraRoot Pull #68](https://github.com/ara-software/AraRoot/pull/68).  This stores the MCTruth values of the theta and phi launch angles in the Report pointer alongside the MCTruth values of the receiving angles of the signal.  The calculation of the launch angles follows the same framework as the receiving angles, which is a simple conversion of the Cartesian vector into theta and phi.

Commit:
Added calculation of launch angles of the RF trajectory from the source vertex.

clark2668 commented on Report.cc:
```diff
@@ -4924,6 +4934,22 @@ void Report::GetAngleAnt(Vector &rec_vector, Position &antenna, double &ant_thet
+    /*
+    2024-07-01 JCF
+    Takes the launch vector of the signal and calculates the theta and phi of the launch vector in the 
+    receiveing antenna's coordinate system.
```
Does it make sense to calculate the _launch_ angle in the frame of the _target_? Wouldn't it make more sense to calculate the launch angle in the frame of the emitter? Or maybe that's what you actually do and the docs are just not quite right?

FlahertyJustinC replied:
Maybe my description was vague.  This calculates the launch angle theta and phi in station-centric coordinates, while launch angle in the Ray Tracer is calculated relative to the target vector.  I've updated the description of the function to hopefully make this more clear.

clark2668 commented on Report.cc:
```diff
@@ -4924,6 +4934,22 @@ void Report::GetAngleAnt(Vector &rec_vector, Position &antenna, double &ant_thet
+    launch_phi = launch_vector.Phi() * TMath::RadToDeg();
+
+}
+
```
Some documentation about units and convention would be good -- it looks like this is reported in degrees? And with what convention, 0 = up, 180 = down, etc.

FlahertyJustinC replied:
Updated the description to have units and conventions for 0 and 180.

clark2668 reviewed (changes requested):
Just a couple questions and requests for more clarifying docs.

FlahertyJustinC committed:
Improved description of Report::GetAngleLaunch() function

Added comments to specify the units and convention of the angles used in the Report::GetAngleLaunch() function.
